code/modules/hydroponics/gene_modder.dm

```diff
@@ -146,13 +150,31 @@
 	ui_interact(user)
 
 /obj/machinery/plantgenes/proc/add_disk(obj/item/disk/plantgene/new_disk, mob/user)
-	if(disk)
-		to_chat(user, "<span class='warning'>A data disk is already loaded into the machine!</span>")
+	if(length(contents) >= disk_capacity)
+		to_chat(user, "<span class='warning'>[src] cannot hold any more disks!</span>")
+		return
+	if(istype(new_disk, /obj/item/storage/box))
+		var/has_disks = FALSE
+		for(var/obj/item/disk/plantgene/D in new_disk.contents)
+			if(length(contents) >= disk_capacity)
+				to_chat(user, "<span class='notice'>You fill [src] with disks.</span>")
```

Comment:
this means that each loop through the box will result in "you fill [src] with disks." I suggest having it change to...

`to_chat(user, "<span class='notice'>You fill [src] with [D].</span>")`

as that will tell you each disks name as it is inserted.

Reply:
This print only happens when you completely fill the DNA manipulator. Printing each disk would be unnecessary spam anyhow, especially considering it's much easier to see which genes you have stored now.